### DISCUSSION TITLE
Allow for different conventional commit scopes based on type of updated dependency

### DISCUSSION THREAD

**Discussion by Ezard** [renovatebot/renovate#10805] [Request Help]
# Problem

I'm aiming to use [semantic-release](https://github.com/semantic-release/semantic-release) for an npm library, and I'd like for it and Renovate to play nicely together

My end goal is something like this:
- when a normal dependency is updated, the patch part of the version of the package should be bumped (e.g. `1.2.3` -> `1.2.4`)
- when a peer dependency is updated (which will only ever be from 1 major version to another), the major part of the version should be bumped (e.g. `1.2.3` -> `2.0.0`)
- when a dev dependency is updated, no part of the version should be bumped

# Proposed solution

Add config options to Renovate to allow the `scope` of conventional commits to be specified based on the type of dependency that was updated, e.g. `chore(deps)`, `chore(devDeps)`, `chore(peerDeps)`
This scope could then be picked up semantic-release's [commit analyser](https://github.com/semantic-release/commit-analyzer)

# Notes

This might be best to be placed into the package manager-specific Renovate config options, since different package managers might have different types of dependencies to each other (e.g. Gradle has `implementation`, `runtime`, etc)

**Comment by Ezard** [comment]
As an addendum to this, if we could also somehow include the path of the dependency file in the scope (to support monorepos, e.g. bump minor version of package `foo` when a normal dependency in `foo/package.json` is updated, by using something like `chore(foo/deps)`) then that'd be awesome too

Not sure if this is a bit far-fetched, but Renovate's manifesto does state `All behavior should be configurable, down to a ridiculous level if necessary` 😄

**Comment by viceice** [comment]
Use `semanticCommitScope` and `matchDepTypes` within package rules.

https://docs.renovatebot.com/configuration-options/#semanticcommitscope
https://docs.renovatebot.com/configuration-options/#matchdeptypes
